### PR TITLE
fix checkbox and late localstorage loading

### PR DESCRIPTION
**Pull request by khalifan-kfan**
PR Checklist (Make sure these options are checked before raising the PR)

- [ ] - Latest `master` branch merged
- [ ] - Descriptive PR title (can be used in release notes)

#### Summary of Changes (What does this PR do?)
- This Pull request fixes a bug realized in the production environment of lazy localstorage loading leading to user ID and token returning as null, especially for new users. It also fixes the checkboxes on medicines by creating a new checkbox component that supports a single entry and will not affect other components accessing the checkbox.

Recreating the error
To emulate a new user, clear your local storage, logout and log back in or use a browser you haven't used before

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Jira Ticket(s) Number
- [SA-87]

## Screenshots/Videos
Not Required

### CODE REVIEW

File: src/axios.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import axios from 'axios';
+const createAxiosInstance = () => {
+  const instance = axios.create();
+  console.log(`Bearer ${JSON.parse(localStorage.getItem('userToken'))}`);
```
Comment (FaithDaka) [inline src/axios.js:4]: Console logs should always be removed before commiting changes

File: src/pages/MedicineTracking/Assessment/PageOne.jsx [hunk 2]
```diff
@@ -33,14 +32,14 @@ const MedicationAssessmentPageOne = () => {
       navigate('/home');
     } catch (err) {
       setLoading(false);
-      alert('Failed to post data')
-   };
-  }
+      alert('Failed to post data');
```
Comment (FaithDaka) [inline src/pages/MedicineTracking/Assessment/PageOne.jsx:35]: Create a page/modal for this alert. This won't be a good UX on the mobile app

File: src/axios.js (hunk 1 above)
Comment (khalifan-kfan) [reply src/axios.js:4]: yes sure

**Comment by khalifan-kfan** [comment]
@FaithDaka, there is a bug for example on the hosted web interface where there is a delay in picking the user ID from the local storage (like its being done throughout the project),  the basis of this fix is to make sure this id is fetched for each API call made in real-time by sending it as a parameter in these API calls such that a new user in the production environment never has to face this issue. sending it from the UI-specific function makes sure of this, just like how the token is picked from local storage and instantiated on each call.
Unless I did not understand the bug properly but the cause was having the slices run and fetch local storage variables on the initial start of the App before they window load This was letter affecting all API calls.

File: src/pages/MedicineTracking/Assessment/PageOne.jsx (hunk 2 above)
Comment (khalifan-kfan) [reply src/pages/MedicineTracking/Assessment/PageOne.jsx:35]: Alright

**Review by KingMike100 — APPROVED** [review]
Thanks Khalifan

**Pull request merged**